### PR TITLE
fix(ValidationError): improve `err.message` for additional properties

### LINKED ISSUE #35: Provide better default error message on invalid properties
### Use Case

I am writing a webpack plugin and want to make sure users don't add extraneous properties to the options. The user should know what specific extraneous property caused the error. Currently it just says

`options should NOT have additional properties`

which does not help the user find out what exactly caused the error.

In the previous release of schema-utils (0.4.x) you would get a message like this which is more helpful:

`options['foo'] is an invalid additional property`

### Implementation

I believe this is caused by removing `errorDataPath: 'property'` from the ajv config as seen here: https://github.com/webpack-contrib/schema-utils/commit/1cbe4ef8bb67171fb8d3515fbbe620de064a188f#diff-813a48919ee26ea79873f7a8e049091fL19

- [ ] I'm open to work on this
  - [ ] I need help/mentorship
- [x] I'm unable to work on this

The `errorDataPath: 'property'` should be added back to the ajv config.

### PR DESCRIPTION
**Pull request by ematipico**
#### `Commit Message Summary (CHANGELOG)`

```bash
feat: added more information to extraneous properties
```

### Type

- [X] Feature

### Issues

- Fixes #35

### SemVer

- [X] Feature (:label: Minor)

### Checklist

- [X] I have signed the CLA
- [X] Lint and unit tests pass with my changes
- [X] I have added tests that prove my fix is effective/works (if needed)
- [ ] I have added necessary documentation (if appropriate)
- [X] Any dependent changes are merged and published in downstream modules

### CODE REVIEW

File: src/ValidationError.js [hunk 1]
```diff
@@ -20,11 +20,29 @@ class ValidationError extends Error {
...
+    const {
+      keyword,
+      message,
+      dataPath,
+      params: { additionalProperty },
+    } = err;
+    const errorMessage = `options${dataPath} ${message}\n`;
+
+    if (keyword === 'additionalProperties') {
+      return `options${err.dataPath} ${
```
Comment (michael-ciniawsky) [inline src/ValidationError.js]: If I remember right the `dataPath` is not defined for `additionalProperties` (`err.dataPath`)? So we might use `additionalProperty` here instead e.g

### Default

```
options${additionalProperty} is NOT a valid options property
```

### Customized

```
options${additionalProperty} is NOT a valid options property (https://github.com/org/repo#options)
```

File: src/ValidationError.js [hunk 2]
```diff
@@ -20,11 +20,29 @@ class ValidationError extends Error {
...
+        err.message
+      }: ${additionalProperty} is extraneous\n`;
```
Comment (michael-ciniawsky) [inline src/ValidationError.js]: If we keep the original `Ajv` `err.message` I would prefer to just add the `additionalProperty`

```
options should NOT have additional properties (${additionalProperty})
```

File: test/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -69,6 +69,18 @@ Array [
 `;
 
 exports[`Error Messages Customized 1`] = `
+Object {
```
Comment (michael-ciniawsky) [inline test/__snapshots__/index.test.js.snap]: This change should not alter the customized snapshots, customizing means to overwrite the `{Error}` messages via the schema (user defined). See the README and https://github.com/webpack-contrib/schema-utils/blob/master/test/fixtures/errors/schema.json for an example. If we change the default `Ajv` messages that's still the default from a user perspective :)

File: test/index.test.js [hunk 4]
```diff
@@ -116,6 +116,7 @@ describe('Error', () => {
         },
         boolean: true,
         instance: new RegExp(''),
+        addedProperty: 'test',
```
Comment (michael-ciniawsky) [inline test/index.test.js]: `additional`

File: test/__snapshots__/index.test.js.snap (hunk 3 above)
Comment (michael-ciniawsky) [reply test/__snapshots__/index.test.js.snap]: This therefore also needs to be tested when using a customized `err.message` in the JSON Schema

**Comment by ematipico** [comment]
I applied the changes! @michael-ciniawsky

**Comment by ematipico** [comment]
Feel free to open it if needed.

**Comment by evilebottnawi** [comment]
@ematipico we rewrite this module for using this inside webpack and loaders/plugins so this problem will be solved, anyway thanks for PR

**Pull request closed without merging**